### PR TITLE
[DOC] Add API migration status page

### PR DESCRIPTION
**Pull request by krisdale**
Add new API migration status page that lists the status of migration support for CUDA APIs.

Additionally includes refactor of DPCT page IDs for simple links from migration status tables.

### CODE REVIEW

**Review by jiejanezhang — APPROVED** [review]

**Comment by zhimingwang36** [comment]
@krisdale *inc files in SYCLomatic are updated,  so please update the PR based on latest files from @wenhuiNi .  Thanks.

File: docs/dev_guide/api-mapping-status/Runtime_and_Driver_API_migration_status.csv [hunk 1]
```diff
@@ -1542,7 +1542,7 @@ cuMemGetAddressRange,NO,
 cuMemGetAllocationGranularity,NO,
 cuMemGetAllocationPropertiesFromHandle,NO,
 cuMemGetHandleForAddressRange,NO,
-cuMemGetInfo,NO,:ref:`DPCT1072`
+cuMemGetInfo,NO,
```
Comment (zhimingwang36) [inline docs/dev_guide/api-mapping-status/Runtime_and_Driver_API_migration_status.csv]: 1. L1545 is duplicated with L1163, Please remove L1545.

File: docs/dev_guide/api-mapping-status/Runtime_and_Driver_API_migration_status.csv [hunk 2]
```diff
@@ -1160,7 +1160,7 @@ cuMemAllocPitch,YES,
 cuMemFree,YES,
 cuMemFreeHost,YES,
 cuMemGetAddressRange,NO,
-cuMemGetInfo,YES,
+cuMemGetInfo,YES,:ref:`DPCT1106`
```
Comment (zhimingwang36) [inline docs/dev_guide/api-mapping-status/Runtime_and_Driver_API_migration_status.csv]: besides DPCT1106,  also add: DPCT1072 for line 1163.

**Review by zhimingwang36 — CHANGES_REQUESTED** [review]
two small change required.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**